subsys/logging/backends/log_backend_net.c

```diff
@@ -92,7 +111,12 @@ static int do_net_init(struct log_backend_net_ctx *ctx)
 
 	local_addr->sa_family = server_addr.sa_family;
 
-	ret = zsock_socket(server_addr.sa_family, SOCK_DGRAM, IPPROTO_UDP);
+	if (ctx->is_tcp) {
```

Comment:
Maybe some explicit error log about misconfiguration could be printed in case  `CONFIG_NET_TCP` is disabled?